### PR TITLE
Extend known HTTP default status messages by RFC's

### PR DESCRIPTION
**Pull request by akzhan**
RFC 2518, RFC 5842, RFC 2324 etc.

Heavily inspired by CPAN HTTP::Status module.

### CODE REVIEW

File: src/http/common.cr [hunk 1]
```diff
@@ -329,21 +332,28 @@ module HTTP
...
-    when 423 then "Locked"
-    when 426 then "Upgrade Required"
+    when 422 then "Unprocessable Entity" # RFC 2518 (WebDAV)
+    when 423 then "Locked"               # RFC 2518 (WebDAV)
+    when 424 then "Failed Dependency"    # RFC 2518 (WebDAV)
+    when 426 then "Upgrade Required"     # RFC 2817
     when 428 then "Precondition Required"
     when 429 then "Too Many Requests"
     when 431 then "Request Header Fields Too Large"
+    when 449 then "Retry with" # unofficial Microsoft
```
Comment (Sija) [inline src/http/common.cr]: `Retry with` -> `Retry With`

Comment (akzhan) [inline src/http/common.cr]: Yes, it looks more consistently. Force updated.

**Comment by ysbaddaden** [comment]
Looks good to me.

**Comment by bcardiff** [comment]
Thanks @akzhan !

**Pull request merged**